napi/index.d.ts

```diff
@@ -0,0 +1,54 @@
+/* auto-generated by NAPI-RS */
+/* eslint-disable */
+export interface CompileOptions {
```

Comment:
Avoid keeping a separate.`.d.ts` that needs to be maintained separate from the source.
Use JSDoc mode for TypeScript https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html

Which can both type check and generate a definitions file.

Reply:
This file is generated by NAPI, based on the Rust lib.rs source. It needs to be in the source code, not sure exactly why, all projects using NAPI do this.